### PR TITLE
Stop device_doctor writing results to stdout

### PR DESCRIPTION
**Pull request by keyonghan**
This is a post cleanup step following https://github.com/flutter/cocoon/pull/1064, which starts writing results to a local file.

Related issue: https://github.com/flutter/flutter/issues/74363

### CODE REVIEW

**Comment by keyonghan** [comment]
Test was fixed in https://github.com/flutter/cocoon/pull/1076, waiting for that PR to land first.

File: device_doctor/lib/src/android_device.dart
Comment (yusufm) [inline device_doctor/lib/src/android_device.dart:120]: Would printing this out to stdout still be helpful? Maybe as a log message?

**Review by yusufm — APPROVED** [review]
LGTM, with a comment for consideration.

File: device_doctor/lib/src/android_device.dart
Comment (keyonghan) [reply device_doctor/lib/src/android_device.dart:120]: Since writeTofile does exactly the same thing as the stdout.write, and from the bot_config side, we are not utilizing the stdout any more, this line doesn't help much.
Would prefer to do the cleanup now. If we need this at any point in the future, we can always add it back.

Comment (godofredoc) [reply device_doctor/lib/src/android_device.dart:120]: Maybe a logging.info? if the properties are unexpected we won't have any way to verify the problem.

Comment (keyonghan) [reply device_doctor/lib/src/android_device.dart:120]: We only have a client side call to the device_doctor, namely no server is running. I am not sure where the logging.info will be logged to?

From bot_config side, if the process fails, we catch the [os_err](https://chrome-internal.googlesource.com/infradata/config/+/refs/heads/master/configs/chromium-swarm/scripts/flutter_mac_android_staging.py#157); if the file parse fails, we catch [io_err](https://chrome-internal.googlesource.com/infradata/config/+/refs/heads/master/configs/chromium-swarm/scripts/flutter_mac_android_staging.py#175) and [value_err](https://chrome-internal.googlesource.com/infradata/config/+/refs/heads/master/configs/chromium-swarm/scripts/flutter_mac_android_staging.py#179). Do we still miss anything?

Comment (godofredoc) [reply device_doctor/lib/src/android_device.dart:120]: The reasoning behind printing or logging the output is that if something goes unexpectedly we can still see the output from /var/log/system or log system in mac even after the file was deleted from the bot.

**Review by godofredoc — APPROVED** [review]
LGTM after adding logging the data.

File: device_doctor/lib/src/android_device.dart
Comment (keyonghan) [reply device_doctor/lib/src/android_device.dart:127]: Added. 
Also fixed a corner case of the getDeviceProperties.

**Comment by godofredoc** [comment]
Awesome, thanks! LGTM.

**Pull request merged**
